ledger-service/http-json/src/main/scala/com/digitalasset/http/json/JsonProtocol.scala

```diff
@@ -357,7 +358,43 @@ object JsonProtocol extends JsonProtocolLow {
       domain.SearchForeverRequest(NonEmptyList((single.convertTo[domain.SearchForeverQuery], 0)))
   }
 
-  implicit val CommandMetaFormat: RootJsonFormat[domain.CommandMeta] = jsonFormat3(
+  implicit val hexStringFormat: JsonFormat[HexString] =
+    jsonFormat[HexString](
+      JsonReader.func2Reader((StringJsonFormat.read _).andThen(HexString.assertFromString)),
+      JsonWriter.func2Writer[HexString](StringJsonFormat.write(_)),
+    )
```

Comment:
If you refactor `MeteringReportEndpoint.stringJsonFormat` here and use it to define this, this becomes a lot nicer.

Reply:
```suggestion
    stringJsonFormat[HexString](
      identity,
      HexString.fromString,
    )
```